### PR TITLE
Add docker

### PR DESCRIPTION
**Pull request by ma7modx**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      DB_HOST: db
+      MIX_ENV: dev
+    volumes:
+        - ./:/app
+    links:
+      - db
+    depends_on:
+      - db
+
+  test:
```
Comment (mostafazh) [inline docker-compose.yml:28]: I dont think we need this in the docker-compose. This is supposed to help others start the app not run unit tests. if someone wants to run tests, they can run the command on the image.

Comment (ma7modx) [reply docker-compose.yml:28]: we're using this in our travis, it has different envs, it has the same image but with different flow

Comment (ma7modx) [reply docker-compose.yml:28]: aslo, won't it useful to run the tests right a way ?

File: .travis.yml [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
+  - sudo rm /usr/local/bin/docker-compose
+  - curl -L https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m` > docker-compose
+  - chmod +x docker-compose
+  - sudo mv docker-compose /usr/local/bin
 
-before_script:
-  - cp config/travis.exs config/test.exs
-  - mix do ecto.create, ecto.migrate
+  - docker-compose --version
+  - docker-compose build test
```
Comment (mostafazh) [inline .travis.yml:14]: docker-compose build local

File: .travis.yml [hunk 3]
```diff
@@ -1,14 +1,27 @@
...
 
 script:
-  - mix test
\ No newline at end of file
+  - docker-compose run test sh test.sh
```
Comment (mostafazh) [inline .travis.yml:17]: docker-compose run local sh test.sh

**Pull request merged**